src/clock.rs

```diff
@@ -64,21 +139,373 @@ impl Strict {
+            uart_clk: Hertz(uart_clk),
+            xtal_freq: Some(Hertz(pll_xtal_freq)),
+            pll_enable: pll_enabled
         }
     }
 }
+
+/// Sets the system clock in the (undocumented) system_core_clock register
+fn system_core_clock_set(value:u32){
+    unsafe { &*pac::HBN::ptr() }.hbn_rsv2.write(|w| unsafe { w
+        .bits(value)
+    })
+}
+
+/// Gets the system clock in the (undocumented) system_core_clock register
+fn system_core_clock_get() -> u32 {
+    unsafe { &*pac::HBN::ptr() }.hbn_rsv2.read().bits()
+}
+
+fn glb_set_system_clk_div(hclkdiv:u8, bclkdiv:u8){
+    // recommended: fclk<=160MHz, bclk<=80MHz
+    // fclk is determined by hclk_div (strange), which then feeds into bclk, hclk and uartclk
+    // glb_reg_bclk_dis isn't in the SVD file, so it isn't generated through svd2rust 
+    // It's only used by this function so define it as a local variable
+    let glb_reg_bclk_dis = 0x40000FFC as * mut u32;
+    unsafe { &*pac::GLB::ptr() }.clk_cfg0.modify(|_, w| unsafe { w
+        .reg_hclk_div().bits(hclkdiv)
+        .reg_bclk_div().bits(bclkdiv)
+    });
+    unsafe { glb_reg_bclk_dis.write_volatile(1) };
+    unsafe { glb_reg_bclk_dis.write_volatile(0) };
+    let currclock = system_core_clock_get();
+    system_core_clock_set(currclock / (hclkdiv as u32 + 1) );
+
+    let mut delay = McycleDelay::new(system_core_clock_get());
+    // This delay used to be 8 NOPS (1/4 us). Might need to be replaced again.
+    delay.try_delay_us(1).unwrap();
+
+    unsafe { &*pac::GLB::ptr() }.clk_cfg0.modify(|_, w| { w
+        .reg_hclk_en().set_bit()
+        .reg_bclk_en().set_bit()
+    });
+    delay.try_delay_us(1).unwrap();
+}
+
+
+fn pds_select_xtal_as_pll_ref(){
+    unsafe { &*pac::PDS::ptr() }.clkpll_top_ctrl.modify(|_, w| {w
+        .clkpll_refclk_sel().set_bit()
+        .clkpll_xtal_rc32m_sel().clear_bit()
+    });
+}
+
+fn pds_power_off_pll(){
+    unsafe { &*pac::PDS::ptr() }.pu_rst_clkpll.modify(|_, w| {w
+        .pu_clkpll_sfreg().clear_bit()
+        .pu_clkpll().clear_bit()
+    });
+
+    unsafe { &*pac::PDS::ptr() }.pu_rst_clkpll.modify(|_, w| {w
+        .clkpll_pu_cp().clear_bit()
+        .clkpll_pu_pfd().clear_bit()
+        .clkpll_pu_fbdv().clear_bit()
+        .clkpll_pu_postdiv().clear_bit()
+    });
+}
+
+/// Power up PLL using C function built into on-chip ROM
+// Powering on the PLL means powering *OFF* the PLL if it's already running.
+// If we're running from flash, this causes a lockup. Running from RAM or ROM is
+// okay though. So there's probably some steps we need to take to do this safely,
+// but the flash peripheral is not documented yet.
+// The easiest solution is to use the C function built into the ROM to do the change.
+#[inline]
+fn pds_power_on_pll_rom(freq: u32) {
+    // Lookup table for ROM function addresses is at 0x21010800
+    // offset for RomDriver_PDS_Power_On_PLL is 88
+    let power_on_pll_lut_entry = (0x21010800 + 88) as * mut u32;
```

Comment:
```suggestion
    let power_on_pll_lut_entry = (0x21010800 + 88) as * mut usize;
```
Since this table stores function pointers, each element should be `usuze`.

Reply:
Agreed, thanks for picking this up